### PR TITLE
Add metrics support for a GCE PD, EC2 EBS & Azure File volumes

### LINKED ISSUE #25840: Expose usage of network volumes via kubelet Summary API
As of now kubelet exports usage of local disk based volumes only. It is useful for users to track usage on network based volumes as well.

### PR DESCRIPTION
**Pull request by vishh**
Added filesystem metrics for GCE PD, EC2 ebs & Azure file

Fixes #25840

cc @thockin

### CODE REVIEW

**Comment by pwittrock** [comment]
Looks like this fails to compile

File: pkg/volume/metrics_statfs.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"errors"
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/volume/util"
+)
+
+var _ MetricsProvider = &metricsStatFS{}
+
+// metricsStatFS represents a MetricsProvider that calculates the used and available
```
Comment (pwittrock) [inline pkg/volume/metrics_statfs.go]: plz update the comment so say it uses statfs to get the aggregate usage metrics for the device.

File: pkg/volume/metrics_statfs.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (pwittrock) [inline pkg/volume/metrics_statfs.go]: nit: update the date if you want

File: pkg/volume/metrics_statfs.go (hunk 1 above)
Comment (vishh) [inline pkg/volume/metrics_statfs.go]: Done.

File: pkg/volume/metrics_statfs.go (hunk 2 above)
Comment (vishh) [inline pkg/volume/metrics_statfs.go]: Done

**Comment by vishh** [comment]
@pwittrock PTAL

**Comment by vishh** [comment]
@k8s-bot test unit github issue #IGNORE

**Comment by pwittrock** [comment]
@k8s-bot test unit github issue #IGNORE

**Comment by alex-mohr** [comment]
Because submit queue is blocked, I'm going to kick off re-tests of the top 5 PRs in the queue, then merge them if they pass.
@k8s-bot test this please: issue #IGNORE

**Pull request merged**